client/app/containers/index.js

```diff
@@ -13,41 +13,96 @@ class App extends React.Component {
         title: '',
         __html: ''
       },
-      errors: []
+      notifications: []
     };
 
-    this.onDirectoryClick = this.onDirectoryClick.bind(this);
+    this.getDirectoryContent = this.getDirectoryContent.bind(this);
     this.traverseToParent = this.traverseToParent.bind(this);
     this.pushNewState = this.pushNewState.bind(this);
-    this.onFileClick = this.onFileClick.bind(this);
-    this.handleError = this.handleError.bind(this);
-    this.removeError = this.removeError.bind(this);
+    this.getFileContent = this.getFileContent.bind(this);
+    this.addNotification = this.addNotification.bind(this);
+    this.removeNotification = this.removeNotification.bind(this);
+    this.subscribeToSocket = this.subscribeToSocket.bind(this);
   }
 
   componentDidMount() {
     const initPath = window.location.pathname;
 
-    this.formNewState(initPath).then(newState => {
-      window.history.replaceState(newState, null, `${initPath}`);
-      window.onpopstate = ({ state }) => this.setState(state);
-      this.setState(newState);
-    }).catch(this.handleError);
+    this.formNewState(initPath)
+      .then(newState => {
+        window.history.replaceState(newState, null, `${initPath}`);
+        window.onpopstate = ({ state }) => this.setState(state);
+        this.setState(newState);
+      })
+      .catch(res => {
+        this.addNotification({
+          text: `${res.status}: ${res.statusText}`,
+          type: 'error'
+        });
+      });
+
+    this.subscribeToSocket();
+  }
+
+  getDirectoryContent(title) {
+    this.pushNewState(`${window.location.pathname}${title}/`);
+  }
+
+  getFileContent(title) {
+    const path = `${window.location.pathname}${title}`;
+    const mapToState = response => this.setState(response);
+
+    api.getFile(path)
+      .then(mapToState)
+      .catch(res => {
+        this.addNotification({
+          text: `${res.status}: ${res.statusText}`,
+          type: 'error'
+        });
+      });
   }
 
-  handleError(res) {
-    const error = {
-      id: Date.now(),
-      text: `${res.status}: ${res.statusText}`
+  subscribeToSocket() {
+    const handleMessage = ({ event }) => {
+      if (event == 'change') {
+        const { currentFile } = this.state;
+
+        this.getFileContent(currentFile.title);
+        this.addNotification({
```

Comment:
will it be hard for 'change' notification to stay there and have 1 at most? So, if you focus away from browser, edit files and go back the badge would stay there and not remove after 2500 (and have only one such notification in list)? 🤔 👼